learn-data-structures-by-building-a-shopping-cart/index.js

```diff
@@ -1,97 +1,207 @@
+const shoppingCartContainer = document.getElementById(
+  "shopping-cart-container"
+);
+const productsContainer = document.getElementById("products-container");
+const dessertCards = document.getElementById("dessert-card-container");
+const shoppingCartBtn = document.getElementById("cart-btn");
+const clearShoppingCartBtn = document.getElementById("clear-cart-btn");
+const totalNumberOfItems = document.getElementById("total-items");
+const cartSubTotal = document.getElementById("subtotal");
+const cartTaxes = document.getElementById("taxes");
+const cartTotal = document.getElementById("total");
+const showHideCartSpan = document.getElementById("show-hide-cart");
+let isCartShowing = false;
+
+/**
+ *
+ * We can decide if we want to provide this products array for them or write steps where the campers
+ * will build it out themselves
+ *
+ */
 const products = [
```

Comment:
We could title case the product names and categories here:

```js
const products = [
  {
    id: 1,
    name: "Vanilla Cupcakes (6 Pack)",
    price: 12.99,
    category: "Cupcake",
  },
  {
    id: 2,
    name: "French Macaroon",
    price: 3.99,
    category: "Macaroon",
  },
  {
    id: 3,
    name: "Pumpkin Cupcake",
    price: 3.99,
    category: "Cupcake",
  },
  {
    id: 4,
    name: "Chocolate Cupcake",
    price: 5.99,
    category: "Cupcake",
  },
  {
    id: 5,
    name: "Chocolate Pretzels (4 Pack)",
    price: 10.99,
    category: "Pretzel",
  },
  {
    id: 6,
    name: "Strawberry Ice Cream",
    price: 2.99,
    category: "Ice Cream",
  },
  {
    id: 7,
    name: "Chocolate Macaroons (4 Pack)",
    price: 9.99,
    category: "Macaroon",
  },
  {
    id: 8,
    name: "Strawberry Pretzel",
    price: 4.99,
    category: "Pretzel",
  },
  {
    id: 9,
    name: "Butter Pecan Ice Cream",
    price: 2.99,
    category: "Ice Cream",
  },
  {
    id: 10,
    name: "Rocky Road Ice Cream",
    price: 2.99,
    category: "Ice Cream",
  },
  {
    id: 11,
    name: "Vanilla Macaroons (5 Pack)",
    price: 11.99,
    category: "Macaroon",
  },
  {
    id: 12,
    name: "Lemon Cupcakes (4 Pack)",
    price: 12.99,
    category: "Cupcake",
  },
];
```

Also, the categories aren't currently shown on the page. I have another suggestion below for adding those to the dessert cards. But I could also see removing the categories entirely if we want fewer steps.

Reply:
Are suggesting this?

```
  {
    Id: 12,
    Name: "Lemon Cupcakes (4 Pack)",
    Price: 12.99,
    Category: "Cupcake",
  },
```

Reply:
No, sorry I wasn't more specific.

I was thinking we could just title case the values for the names and categories, not the keys themselves. Something like this:

```
  {
    id: 12,
    name: "Lemon Cupcakes (4 Pack)",
    price: 12.99,
    category: "Cupcake",
  },
```

Reply:
I updated the product names